### DISCUSSION TITLE
Why does envkey-vscode display information in the Problems panel?

### DISCUSSION THREAD

**Discussion by miquelbrazil** [envkey/envkey#72] [Q&A] [answered]
I'm curious about the decision to display context information about EnvKey variables in the Problems panel when using the VSCode extension. Is this the only way to provide this information to the hover/peek window when hovering over a symbol in the editor? 

This makes sense for those variables declared in the code, but EnvKey can't find a value for them (`EnvKey: ENV_VAR not set in current environment`). But I'm not sure why variables that are properly set have error underlines and also appear in the Problems panel.

I realize it's a small thing aesthetically, but it's been getting in the way of my cognitive flow when reviewing or refactoring code to the point that I'm not sure if there's much benefit to using it. Is this by design? What's the logic behind doing it this way? Will it change at some point in the future?

**Comment by danenania** [comment, marked as answer]
There unfortunately doesn't seem to be a way to add VSCode diagnostics, which provide the underlining/hover behavior, without also having them show in the Problems panel. I agree it's annoying and would be happy to change it if VSCode offers an alternative in the future.

**Reply by danenania** [reply]
Expanding a little more, we are already using the `vscode.DiagnosticSeverity.Information` severity level for existing variables, which is specifically described as `Something to inform about but not a problem.` So really VSCode should not be showing these as problems--not sure why it does.

**Reply by miquelbrazil** [reply]
Thanks for this explanation. As I said, I was more curious than anything else. I should mention how dope I think it is that you provide this level of integration across the service.

Your response led me to discover the `Toggle Infos` option in the Problems pane's Filter menu. Since I use a Configuration class at the application layer, I typically declare the use of all my environment variables in a single config file when the app processes a request (I'm using EnvKey with a PHP web application). If VSCode allows filters to be applied to specific files or file types, I'll have a pretty good workaround for the time being.

Thanks again.
